query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryRequestUtil.java

```diff
@@ -68,4 +71,23 @@ public static Expression createNullNumberLiteralExpression() {
                 .setValue(Value.newBuilder().setValueType(ValueType.NULL_NUMBER)))
         .build();
   }
+
+  public static boolean isDateTimeFunction(Expression expression) {
+    return expression.getValueCase() == ValueCase.FUNCTION
+        && expression.getFunction().getFunctionName().equals("dateTimeConvert");
+  }
+
+  public static boolean isTimeColumn(String columnName) {
```

Comment:
This isn't a safe assumption, we should not be hardcoding column names.

Reply:
> I think I'd still recommend starting from the current API and working down from there even so, so we don't run into surprises.

Although we are down the other path, just curious, do you mean starting with a new api at qs layer?

Reply:
I can take the timeColumn name as config, but I ran into a doubt, the existing view definition map both startTime and endTime attribute. But there is only a single field for time column in the view Definition config and in the code [here](https://github.com/hypertrace/query-service/blob/main/query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java#L249), is `endTime` attribute not used in the queryRequest?

Reply:
I believe there's a separate part of the config, aside from the columns, that defines the time column. That definitely exists on the view creation side (since time is used internally by pinot for optimization) and on the gateway side to do the initial translation, but I'm like 80% sure we define it in the QS config too.